### PR TITLE
Add Sort to Structured Content

### PR DESCRIPTION
**Pull request by cgoncas**

### CODE REVIEW

File: modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/sort/SortParser.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+		return Optional.of(new SortKey(fieldName, asc));
+	}
+
+	protected boolean isAsc(String orderBy) {
+		if (orderBy == null) {
+			return _DEFAULT_ASC;
+		}
+
+		if (com.liferay.portal.kernel.util.StringUtil.equals(
```
Comment (nhpatt) [inline modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/sort/SortParser.java]: Argh, damn so many StringUtil with incomplete methods. I would use the normal equals and toLowerCase.

File: modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/test/java/com/liferay/structured/content/apio/internal/architect/sort/SortParserTest.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+import org.assertj.core.api.AbstractThrowableAssert;
+import org.assertj.core.api.Assertions;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * @author Cristina González
+ */
+public class SortParserTest {
```
Comment (nhpatt) [inline modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/test/java/com/liferay/structured/content/apio/internal/architect/sort/SortParserTest.java]: 👏

File: modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java [hunk 3]
```diff
@@ -393,14 +416,23 @@ private String _getLocalizedString(
...
+			ThemeDisplay themeDisplay, Sort sort)
 		throws PortalException {
 
+		OrderByComparator<JournalArticle> orderByComparator = null;
+
+		if (sort.hasValue()) {
+			Optional<String> value = sort.getValue();
+
+			orderByComparator = _getJournalArticleOrderByComparator(
+				_sortParser.parse(value.get()));
```
Comment (nhpatt) [inline modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java]: This is the only thing I don't like of the API... that you have to know that the value returned from the Sort has to be parsed by an internal SortParser... I don't know how (maybe sacrificing the component nature of SortParser), I would prefer an API that returned a Sort object with the values for asc, key... directly without parsing.

File: modules/apps/headless-apio/structured-content/structured-content-apio-api/src/main/java/com/liferay/structured/content/apio/architect/sort/Sort.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.structured.content.apio.architect.sort;
+
+import java.util.Optional;
+
+/**
+ * Instances of this interface represent a Sort for sort structured
```
[the remark this answers is not in the document]
Comment (cgoncas) [reply modules/apps/headless-apio/structured-content/structured-content-apio-api/src/main/java/com/liferay/structured/content/apio/architect/sort/Sort.java]: Good catch :)

**Pull request closed without merging**